scm-ui/ui-webapp/src/containers/Logout.tsx

```diff
@@ -24,50 +24,48 @@
 import React from "react";
 import { connect } from "react-redux";
 import { WithTranslation, withTranslation } from "react-i18next";
-import { Redirect } from "react-router-dom";
 
-import { getLogoutFailure, isAuthenticated, isLogoutPending, isRedirecting, logout } from "../modules/auth";
+import { getLogoutFailure, logout } from "../modules/auth";
 import { ErrorPage, Loading } from "@scm-manager/ui-components";
 import { getLogoutLink } from "../modules/indexResource";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import { compose } from "redux";
 
-type Props = WithTranslation & {
-  authenticated: boolean;
-  loading: boolean;
-  redirecting: boolean;
-  error: Error;
-  logoutLink: string;
+type Props = RouteComponentProps &
+  WithTranslation & {
+    error: Error;
+    logoutLink: string;
 
-  // dispatcher functions
-  logout: (link: string) => void;
-};
+    // dispatcher functions
+    logout: (link: string) => void;
+  };
 
 class Logout extends React.Component<Props> {
   componentDidMount() {
-    this.props.logout(this.props.logoutLink);
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        if (this.props.logoutLink) {
+          this.props.logout(this.props.logoutLink);
+          resolve(this.props.history.push("/login"));
+        }
+      });
+    });
```

Comment:
What does this code do? It creates a promise, but nobody is waiting for it. It uses setTimeout, but without a timeout value. Why not:

```javascript
if (this.props.logoutLink) {
  this.props.logout(this.props.logoutLink);
  resolve(this.props.history.push("/login"));
}
```

Without promise and without setTimeout?